src/os/ObjectStore.h

```diff
@@ -1716,71 +1716,45 @@ class ObjectStore {
     static void generate_test_instances(list<Transaction*>& o);
   };
 
-  struct C_DeleteTransaction : public Context {
-    ObjectStore::Transaction *t;
-    C_DeleteTransaction(ObjectStore::Transaction *tt) : t(tt) {}
-    void finish(int r) {
-      delete t;
-    }
-  };
-  template<class T>
-  struct C_DeleteTransactionHolder : public Context {
-    ObjectStore::Transaction *t;
-    T obj;
-    C_DeleteTransactionHolder(ObjectStore::Transaction *tt, T &obj) :
-      t(tt), obj(obj) {}
-    void finish(int r) {
-      delete t;
-    }
-  };
-
   // synchronous wrappers
   unsigned apply_transaction(Sequencer *osr, Transaction& t, Context *ondisk=0) {
-    list<Transaction*> tls;
-    tls.push_back(&t);
+    vector<Transaction> tls;
```

Comment:
@cbodley I am halfway of changing apply_transaction() and I was thinking should we really want to change the apply_transaction() ??
With the current implementation is almost no overhead , but, if we go with move route , there will be two std::move()..Moving transaction has some overhead (at least copying native types)...Let me know what you think..